### PR TITLE
refactor: move all proto files to a standalone module

### PR DESCRIPTION
**Pull request by imotai**
Based on the requirements outlined in issue #3073, I believe moving all proto files to a standalone module would provide a better experience for library users.
By the way. I am one of the users.

### CODE REVIEW

**Review by wjones127 — COMMENTED** [review]
👍 Creating a new package makes sense.

However, I think instead of using a `build.rs` script, the ticket asks us to check in the generated files. So I think the acceptance criteria here are:

1. No `build.rs` should need `protoc`. All the `protoc` files should be generated manually when the files are changed.
    * As part of this, we should be able to remove all CI job lines that require protoc. For example: https://github.com/lancedb/lance/blob/638eaf4e9a31c4d4f9f311667e04bfcc56224313/.github/workflows/rust.yml#L70 
2. There needs to be a CI job to validate that the generated protobuf definitions are in sync with the `.proto` files. One possibility is running the protobuf build script and then using `git` to see if there are any changes.

**Comment by imotai** [comment]
1. Done
2. I believe we still need build.rs to regenerate the code file whenever the protobuf file changes

File: rust/lance-proto/src/lib.rs
Comment (wjones127) [inline rust/lance-proto/src/lib.rs:1]: You might have to add some lines to allow certain lints in generated files:

```rust
#[allow(clippy::___)]
```

**Comment by wjones127** [comment]
> I believe we still need build.rs to regenerate the code file whenever the protobuf file changes

I think as long as there is a `build.rs`, it will mean there is a build time dependency on `protoc`.

Here is a good inspiration from another project that demonstrates putting generating the protobuf artifacts in a separate script: https://github.com/apache/arrow-rs/pull/3927

**Comment by loloxwg** [comment]
Is this pull request still in progress?

**Comment by wjones127** [comment]
Weird that this was closed. It has zero commits.

@imotai do you still plan on working on this? If so, feel free to open a fresh PR.

**Pull request merged**